### PR TITLE
Statically link SDL2

### PR DESCRIPTION
**Pull request by Zer0-One**
This only works if your system provides a static SDL2 built with `-fPIC`. Currently that doesn't include any distro I know about, so this fails on Linux. The better solution is probably to bring SDL2 into the build, or at the very least to just tell users they have to provide an appropriately-built lib.

EDIT: Switched to static linking on MSVC only. I don't like it, but i guess Windows gonna Windows, so w/e

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -61,15 +61,15 @@ jobs:
...
+      run: vcpkg install sdl2[core]:x64-windows-static
 
     # setup vs env
     - uses: ilammy/msvc-dev-cmd@v1
       with:
         arch: amd64
 
     - name: Configure CMake
-      run: cmake -G Ninja -B ${{github.workspace}}/build -DCMAKE_BUILD_TYPE=${{env.BUILD_TYPE}} -DCMAKE_TOOLCHAIN_FILE=C:/vcpkg/scripts/buildsystems/vcpkg.cmake
+      run: cmake -G Ninja -B ${{github.workspace}}/build -DCMAKE_BUILD_TYPE=${{env.BUILD_TYPE}} -DCMAKE_TOOLCHAIN_FILE=C:/vcpkg/scripts/buildsystems/vcpkg.cmake -DVCPKG_TARGET_TRIPLET=x64-windows-static
```
Comment (Green-Sky) [inline .github/workflows/ci.yml:72]: do we *need* this?, or does it not find it otherwise?

File: src/CMakeLists.txt [hunk 2]
```diff
@@ -14,15 +14,19 @@ target_link_libraries(video PUBLIC mumble_plugin)
...
-	target_include_directories(video PUBLIC "${SDL2_INCLUDE_DIR}")
-	target_link_libraries(video PUBLIC ${SDL2_LIBRARY})
+	target_link_libraries(video
+		PUBLIC
+			SDL2::SDL2main
+			SDL2::SDL2
+	)
 endif()
 
+target_include_directories(video PUBLIC "${SDL2_INCLUDE_DIR}")
```
Comment (Green-Sky) [inline src/CMakeLists.txt]: if the sdl targets are created properly, this should not be necessary.

**Comment by Green-Sky** [comment]
also the windows release build failed :thinking:

File: .github/workflows/ci.yml (hunk 1 above)
Comment (Zer0-One) [reply .github/workflows/ci.yml:72]: we need it

**Review by Green-Sky — APPROVED** [review]

**Pull request merged**